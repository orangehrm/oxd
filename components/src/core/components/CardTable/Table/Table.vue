<template>
  <div class="oxd-table" role="table">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'oxd-card-table-container',

  computed: {
    classes(): object {
      return {
        'oxd-table': true,
      };
    },
  },
});
</script>

<style src="./table.scss" lang="scss"></style>
