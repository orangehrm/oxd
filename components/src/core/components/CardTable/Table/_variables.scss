$oxd-table-font-size: 0.75rem !default;
$oxd-table-header-font-weight: 700 !default;
$oxd-table-font-weight: 400 !default;
$oxd-table-font-family: $oxd-font-family !default;
$oxd-table-font-color: $oxd-interface-gray-darken-1-color !default;

$oxd-table-row-hover-bg-color: $oxd-background-light-gray-color !default;
$oxd-table-row-hover-font-color: $oxd-interface-gray-darken-2-color !default;
$oxd-table-row-border: $oxd-border !default;
$oxd-table-row-border-color: $oxd-interface-gray-lighten-2-color !default;

$oxd-table-body-strip-color: $oxd-background-white-shadow-color !default;

$oxd-table-tr-bg-color: $oxd-interface-gray-color !default;
$oxd-table-tr-padding: 25px !default;

$oxd-table-cell-padding: 5px !default;
$oxd-table-row-left-padding: 5px !default;
$oxd-table-row-right-padding: 5px !default;
$oxd-card-table-row-gap: 5px !default;

$oxd-table-loader-height: 400px !default;

$oxd-table-sort-dropdown-padding: 0.125rem !default;
$oxd-table-sort-dropdown-top-margin: 1.5rem !default;
$oxd-table-sort-dropdown-height: 4rem !default;
$oxd-table-sort-dropdown-width: 6rem !default;
$oxd-table-sort-dropdown-font-size: 12px !default;
$oxd-table-sort-dropdown-font-weight: 600 !default;
$oxd-table-sort-dropdown-font-color: $oxd-interface-gray-darken-1-color !default;
