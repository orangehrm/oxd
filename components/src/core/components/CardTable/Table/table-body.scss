@import '../../../../styles';
@import 'variables';

.oxd-table-body {
  font-weight: $oxd-table-font-weight;
  display: grid;
  grid-gap: $oxd-card-table-row-gap;
}
