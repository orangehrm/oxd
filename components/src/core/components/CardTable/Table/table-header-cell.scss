@import '../../../../styles';
@import 'variables';

.oxd-table-header-cell {
  display: flex;
  flex-direction: row;
  align-items: center;
  &-sorticon {
    margin-left: 10px;
    font-size: inherit;
  }
}
