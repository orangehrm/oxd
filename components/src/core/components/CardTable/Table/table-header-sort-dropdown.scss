// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import '../../../../styles';
@import 'variables';

.oxd-table-header-sort {
  position: relative;

  &-icon {
    margin-left: 10px;
    font-size: inherit;
  }

  &-dropdown {
    right: 0;
    display: block;
    position: absolute;
    overflow: hidden auto;
    top: $oxd-table-sort-dropdown-top-margin;
    max-height: $oxd-table-sort-dropdown-height;
    width: $oxd-table-sort-dropdown-width;
    box-shadow: $oxd-dropdown-shadow;
    border: $oxd-dropdown-border;
    border-radius: $oxd-dropdown-border-radius;
    background-color: $oxd-dropdown-background-color;
    z-index: $oxd-dropdown-z-index;

    ul {
      list-style: none;
      padding: $oxd-dropdown-padding;
      margin: 0;
    }

    &-item {
      cursor: pointer;
      text-align: right;
      font-size: $oxd-table-sort-dropdown-font-size;
      font-weight: $oxd-table-sort-dropdown-font-weight;
      color: $oxd-table-sort-dropdown-font-color;
      padding: $oxd-dropdown-padding;
      border-radius: $oxd-dropdown-border-radius;
      span {
        margin-left: 5px;
      }
      &:hover,
      &.--selected {
        background-color: $oxd-dropdown-item-hover-color;
      }
    }
  }
}
