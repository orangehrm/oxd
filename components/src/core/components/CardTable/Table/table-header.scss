@import '../../../../styles';
@import 'variables';

.oxd-table-header {
  font-weight: $oxd-table-header-font-weight;
  display: flex;
  flex-direction: column;
}
