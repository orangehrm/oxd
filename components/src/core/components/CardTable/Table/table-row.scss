@import '../../../../styles';
@import 'variables';

.oxd-table-row {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  height: 100%;
  padding-left: $oxd-table-row-left-padding;
  padding-right: $oxd-table-row-right-padding;
}

.oxd-table-body > .oxd-table-row {
  &:hover {
    color: $oxd-table-row-hover-font-color;
    background-color: $oxd-table-row-hover-bg-color;
  }

  &--with-border {
    border-top: $oxd-table-row-border $oxd-table-row-border-color;
  }
}

.oxd-table-row--clickable {
  cursor: pointer;
}
