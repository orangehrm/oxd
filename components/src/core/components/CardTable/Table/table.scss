@import '../../../../styles';
@import 'variables';

.oxd-table {
  color: $oxd-table-font-color;
  font-family: $oxd-table-font-family;
  font-size: $oxd-table-font-size;
  display: flex;
  flex-direction: column;

  &-loader {
    min-height: $oxd-table-loader-height;
    width: 100%;
    z-index: 999;
    display: flex;
    justify-content: center;
    align-items: center;
    & .oxd-loading-spinner-container {
      background-color: $oxd-white-color;
    }
  }
}

.oxd-padding-cell {
  padding: $oxd-table-cell-padding;
}

.oxd-table-cell {
  overflow-wrap: break-word;
  word-break: break-word;
}
