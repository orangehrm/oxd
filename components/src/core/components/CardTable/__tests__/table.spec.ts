import {mount} from '@vue/test-utils';
import Table from '@orangehrm/oxd/core/components/CardTable/Table/Table.vue';

describe('CardTable > Table.vue', () => {
  it('renders OXD CardTable > Table', () => {
    const wrapper = mount(Table, {
      props: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
