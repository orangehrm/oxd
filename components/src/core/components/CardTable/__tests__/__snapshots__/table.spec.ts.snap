// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable > Table.vue renders OXD CardTable > Table 1`] = `<div class="oxd-table" role="table"></div>`;
